src/game/common/skirmishbattlehonors.cpp

```diff
@@ -27,3 +39,617 @@ int SkirmishBattleHonors::Get_Endurance_Medal(Utf8String name, int level) const
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_CHALLENGE, &row, &column, 0, 0);
+    } else if (bronze) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Challenge_Bronz");
+        Insert_Battle_Honor(listbox, image, 1, CAMPAIGN_CHALLENGE, &row, &column, 0, 0);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Challenge_Bronz");
+        Insert_Battle_Honor(listbox, image, 0, CAMPAIGN_CHALLENGE, &row, &column, 0, 0);
+    }
+
+    image = g_theMappedImageCollection->Find_Image_By_Name("HonorAirWing");
+    Insert_Battle_Honor(listbox, image, (char)(honors >> 8) & 1, AIR_WING, &row, &column, 0, 0);
+
+    image = g_theMappedImageCollection->Find_Image_By_Name("HonorBattleTank");
+    Insert_Battle_Honor(listbox, image, (char)(honors >> 7) & 1, BATTLE_TANK, &row, &column, 0, 0);
+
+    Gadget_List_Box_Add_Entry_Image(listbox, 0, 2, 0, 10, 10, 1, -1);
+
+    row = 3;
+    column = 0;
+
+    // Endurance: beat every map
+    bronze = true;
+    silver = true;
+    gold = true;
+    for (auto it = g_theMapCache->begin(); it != g_theMapCache->end(); it++) {
+        if (it->second.m_isOfficial && it->second.m_isMultiplayer) {
+            int beatHard = BattleHonors->Get_Endurance_Medal(it->first, 4);
+            if (!beatHard) {
+                gold = false;
+                int beatMedium = BattleHonors->Get_Endurance_Medal(it->first, 3);
+                if (!beatMedium) {
+                    silver = false;
+                    int beatEasy = BattleHonors->Get_Endurance_Medal(it->first, 2);
+                    if (!beatEasy) {
+                        bronze = false;
+                        break;
+                    }
+                }
+            }
+        }
+    }
+    if (gold) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Endurance_G");
+        Insert_Battle_Honor(listbox, image, 1, ENDURANCE, &row, &column, 0, 0);
+    } else if (silver) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Endurance_S");
+        Insert_Battle_Honor(listbox, image, 1, ENDURANCE, &row, &column, 0, 0);
+    } else if (bronze) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Endurance_B");
+        Insert_Battle_Honor(listbox, image, 1, ENDURANCE, &row, &column, 0, 0);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Endurance_B");
+        Insert_Battle_Honor(listbox, image, 0, ENDURANCE, &row, &column, 0, 0);
+    }
+
+    // Apocalypse: build all superweapons
+    image = g_theMappedImageCollection->Find_Image_By_Name("Apocalypse");
+    Insert_Battle_Honor(listbox, image, (char)(honors >> 0x11) & 1, APOCALYPSE, &row, &column, 0, 0);
+
+    // Blitz: win in (x) minutes or less
+    if ((honors & 0x4000) != 0) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorBlitz5");
+        Insert_Battle_Honor(listbox, image, 1, BLITZ_5, &row, &column, 0, 0);
+    } else if ((honors & 0x8000u) == 0) {
```

Comment:
Perhaps flip condition to `(honors & BLITZ_10) != 0` and list the enabled image first, and disabled image last, like in the other cases.